lib/localisation/src/filtering/kalman_filter.rs

```diff
@@ -0,0 +1,144 @@
+use nalgebra::{DMatrix, DVector};
+
+pub struct KalmanFilter {
```

Comment:
Would be good to write a Rust doc comment briefly explaining what the Kalman Filter is doing and why we use it